kernel/src/kernel.rs

```diff
@@ -361,6 +361,21 @@ impl Kernel {
         }
     }
 
+    fn service_interrupts<C: Chip>(&self, chip: &C) {
+        chip.service_pending_interrupts();
+    }
+
+    fn service_deferred_call(&self) {
+        DeferredCall::service_next_pending();
+    }
```

Comment:
Service a single deferred call at a time. This keeps the original behaviour: service deferred calls while there are no interrupts, since `should_do_kernel_work()` prioritises interrupts over deferred calls by default.